segfault when building in a very narrow terminal

### Contributing guidelines and issue reporting guide

- [x] I've read the [contributing guidelines](https://github.com/moby/buildkit/blob/master/.github/CONTRIBUTING.md) and wholeheartedly agree. I've also read the [issue reporting guide](https://github.com/moby/buildkit/blob/master/.github/issue_reporting_guide.md).

### Well-formed report checklist

- [x] I have found a bug that the documentation does not mention anything about my problem
- [x] I have found a bug that there are no open or closed issues that are related to my problem
- [x] I have provided version/information about my environment and done my best to provide a reproducer

### Description of bug

## Bug description

I am building a container in a terminal session in my editor. A pane expands, shrinking my terminal pane. It's now very narrow - and the docker build crashes with a panic.

I can consistently reproduce this. I get the following stack trace if the terminal is 10 columns wide or narrower.

```panic: runtime error: makeslice: len out of range
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x60 pc=0xe979b4]

goroutine 98 [running]:
github.com/moby/buildkit/util/progress/progressui.(*trace).displayInfo(0xc000a882a0)
	github.com/moby/buildkit@v0.21.0/util/progress/progressui/display.go:809 +0x134
github.com/moby/buildkit/util/progress/progressui.(*consoleDisplay).done(0xc000608360)
	github.com/moby/buildkit@v0.21.0/util/progress/progressui/display.go:228 +0x46
panic({0x23b2b00?, 0x2b8ab50?})
	runtime/panic.go:791 +0x132
github.com/tonistiigi/vt100.NewVT100(0x6, 0xfffffffffffffffc)
	github.com/tonistiigi/vt100@v0.0.0-20240514184818-90bafcd6abab/vt100.go:160 +0x109
github.com/moby/buildkit/util/progress/progressui.(*trace).update(0xc000a882a0, 0xc002ad6660, 0x6)
	github.com/moby/buildkit@v0.21.0/util/progress/progressui/display.go:665 +0x827
github.com/moby/buildkit/util/progress/progressui.(*consoleDisplay).update(0xc000608360, 0xc002ad6660)
	github.com/moby/buildkit@v0.21.0/util/progress/progressui/display.go:214 +0x48
github.com/moby/buildkit/util/progress/progressui.Display.UpdateFrom({{0x2bbab68?, 0xc000608360?}}, {0x2bba700, 0xc00060e1e0}, 0xc0001c82a0)
	github.com/moby/buildkit@v0.21.0/util/progress/progressui/display.go:110 +0x2f3
github.com/docker/buildx/util/progress.NewPrinter.func1()
	github.com/docker/buildx@v0.23.0/util/progress/printer.go:134 +0x1a8
created by github.com/docker/buildx/util/progress.NewPrinter in goroutine 16
	github.com/docker/buildx@v0.23.0/util/progress/printer.go:121 +0x2b1
```

10 columns is very narrow indeed. Here's what it looks like - the terminal is in the bottom middle:
![Image](https://github.com/user-attachments/assets/66e00f18-dc06-4dcf-a4bb-1d66087b6bbb)

## Reproduction

I'm on linux, I don't know if that matters. 

I am able to reproduce in terminals outside of my editor.

There are two variants:
1. Try to start a build in a terminal that is  _already_ 10 columns or narrower. I get this:
```
 docker buildx build .
panic: invalid dim (6, 0)
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x60 pc=0x1011fb4]

goroutine 35 [running]:
github.com/moby/buildkit/util/progress/progressui.(*trace).displayInfo(0xc000348f00)
	github.com/moby/buildkit@v0.21.0/util/progress/progressui/display.go:809 +0x134
github.com/moby/buildkit/util/progress/progressui.(*consoleDisplay).done(0xc0003b8660)
	github.com/moby/buildkit@v0.21.0/util/progress/progressui/display.go:228 +0x46
panic({0x23d2f20?, 0xc00061d980?})
	runtime/panic.go:791 +0x132
github.com/tonistiigi/vt100.NewVT100(0x23f6540?, 0x0)
	github.com/tonistiigi/vt100@v0.0.0-20240514184818-90bafcd6abab/vt100.go:149 +0x2c5
github.com/moby/buildkit/util/progress/progressui.(*trace).update(0xc000348f00, 0xc000611bc0, 0xa)
	github.com/moby/buildkit@v0.21.0/util/progress/progressui/display.go:665 +0x827
github.com/moby/buildkit/util/progress/progressui.(*consoleDisplay).update(0xc0003b8660, 0xc000611bc0)
	github.com/moby/buildkit@v0.21.0/util/progress/progressui/display.go:214 +0x48
github.com/moby/buildkit/util/progress/progressui.Display.UpdateFrom({{0x2d90a38?, 0xc0003b8660?}}, {0x2d90308, 0xc0006525f0}, 0xc000540070)
	github.com/moby/buildkit@v0.21.0/util/progress/progressui/display.go:110 +0x2f3
github.com/docker/buildx/util/progress.NewPrinter.func1()
	github.com/docker/buildx/util/progress/printer.go:134 +0x1a8
created by github.com/docker/buildx/util/progress.NewPrinter in goroutine 1
	github.com/docker/buildx/util/progress/printer.go:121 +0x2b1

```

2. Starts a build, then shrink the window; I get the other stack trace I mentioned earlier.

## Version information


```bash
docker buildx version && docker buildx inspect
github.com/docker/buildx v0.23.0 28c90ea
Name:          default
Driver:        docker
Last Activity: 2025-05-09 23:07:59 +0000 UTC

Nodes:
Name:             default
Endpoint:         default
Status:           running
BuildKit version: v0.21.0
Platforms:        linux/amd64, linux/amd64/v2, linux/amd64/v3, linux/amd64/v4, linux/386
Labels:
 org.mobyproject.buildkit.worker.moby.host-gateway-ip: 172.17.0.1
```

--- a/util/progress/progressui/display.go
+++ b/util/progress/progressui/display.go
@@ -640,7 +640,11 @@ func (t *trace) update(s *client.SolveStatus, termWidth int) {
 					subVtxs: make(map[digest.Digest]client.Vertex),
 				}
 				if t.modeConsole {
-					group.term = vt100.NewVT100(termHeight, termWidth-termPad)
+					w := termWidth - termPad
+					if w <= 0 {
+						w = 1
+					}
+					group.term = vt100.NewVT100(termHeight, w)
 				}
 				t.groups[v.ProgressGroup.Id] = group
 				t.byDigest[group.Digest] = group.vertex
@@ -661,7 +665,11 @@ func (t *trace) update(s *client.SolveStatus, termWidth int) {
 				intervals:     make(map[int64]interval),
 			}
 			if t.modeConsole {
-				t.byDigest[v.Digest].term = vt100.NewVT100(termHeight, termWidth-termPad)
+				w := termWidth - termPad
+				if w <= 0 {
+					w = 1
+				}
+				t.byDigest[v.Digest].term = vt100.NewVT100(termHeight, w)
 			}
 		}
 		t.triggerVertexEvent(v)
@@ -998,6 +1006,9 @@ func (disp *ttyDisplay) print(d displayInfo, width, height int, all bool) {
 		out = align(out, disp.desc, width-1)
 	} else {
 		out = align(out, "", width)
+	}
+	if len(out) > width {
+		out = out[:width]
 	}
 	fmt.Fprintln(disp.c, out)
 	lineCount := 0

